### PR TITLE
4 - add get article by id

### PR DESCRIPTION
**Pull request by kneeshuh**
Add get article by id functions, tests, and addition to endpoints.json

### CODE REVIEW

File: models/articles.models.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const db = require('../db/connection')
+
+exports.fetchArticleById = (article_id) => {
+    return db.query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+    .then(({ rows }) => {
+        if (rows.length === 0) {
+            return Promise.reject({status: 404, msg: 'not found'})
+        }
+        return rows
```
Comment (nc-christian-dutton) [inline models/articles.models.js:9]: return rows[0] as you only have one object to return

File: __tests__/app.test.js [hunk 2]
```diff
@@ -45,4 +45,39 @@ describe('/api', () => {
...
+            article.forEach((article) => {
+                expect(article).toHaveProperty('article_id')
+                expect(article).toHaveProperty('title')
+                expect(article).toHaveProperty('topic')
+                expect(article).toHaveProperty('author')
+                expect(article).toHaveProperty('body')
+                expect(article).toHaveProperty('created_at')
+                expect(article).toHaveProperty('votes')
+                expect(article).toHaveProperty('article_img_url')
+            })
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:66]: shouldn't need to forEach here if you return rows[0] in the model, it should just be one article rather than an array of them. 

I would be more specific with the testing, at a minimum I would check that article.article_id is 1 as that proves you have the article you have asked for. We should be able to assert exactly what the article looks like

File: __tests__/app.test.js [hunk 3]
```diff
@@ -45,4 +45,39 @@ describe('/api', () => {
...
+        .then(({ body }) => {
+            expect(body.msg).toBe('not found')
+        })
+    })
+    test('400: send appropriate error status and message when given invalid article_id', () => {
+        return request(app).get('/api/articles/not-an-article-id')
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe('bad request')
+        })
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:81]: good error testing! but I would make the articles number much bigger, remember you are testing with a test database, the actual database might have more articles? or what if you add more? change it to a huge number :)
